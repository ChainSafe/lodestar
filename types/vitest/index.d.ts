// eslint-disable-next-line import/no-extraneous-dependencies, @typescript-eslint/no-unused-vars
import * as vitest from "vitest";

interface CustomMatchers<R = unknown> {
  toBeValidEpochCommittee(opts: {committeeCount: number; validatorsPerCommittee: number; slotsPerEpoch: number}): R;
  /**
   * @deprecated
   * We highly recommend to not use this matcher instead use detail test case
   * where you don't need message to explain assertion
   *
   * @example
   * ```ts
   * it("should work as expected", () => {
   *   const a = 1;
   *   const b = 2;
   *   expect(a).toBeWithMessage(b, "a must be equal to b");
   * });
   * ```
   * can be written as:
   * ```ts
   * it("a should always same as b", () => {
   *   const a = 1;
   *   const b = 2;
   *   expect(a).toBe(b);
   * });
   * ```
   * */
  toBeWithMessage(expected: unknown, message: string): R;
  /**
   * @deprecated
   * We highly recommend to not use this matcher instead use detail test case with .toEqual
   * where you don't need message to explain assertion
   * */
  toEqualWithMessage(expected: unknown, message: string): R;
}

interface CustomAsymmetricMatchers<R = unknown> extends CustomMatchers<R> {
  /**
   * Non-asymmetric matcher already exists, we just need to add asymmetric version
   */
  toSatisfy(func: (received: unknown) => boolean): R;
}

declare module "vitest" {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  interface Assertion<T = any> extends CustomMatchers<T> {}
  interface AsymmetricMatchersContaining extends CustomAsymmetricMatchers {}
}
