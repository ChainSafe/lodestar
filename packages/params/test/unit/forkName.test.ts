import {describe, it, expect} from "vitest";
import {
  ForkName,
  forkAll,
  forkBlobs,
  forkExecution,
  forkLightClient,
  forkWithdrawals,
  highestFork,
  lowestFork,
} from "../../src/forkName.js";

describe("forkName", () => {
  it("should have valid allForks", () => {
    expect(forkAll).toMatchSnapshot();
  });

  it("should have valid execution forks", () => {
    expect(forkExecution).toMatchSnapshot();
  });

  it("should have valid lightclient forks", () => {
    expect(forkLightClient).toMatchSnapshot();
  });

  it("should have valid withdrawal forks", () => {
    expect(forkWithdrawals).toMatchSnapshot();
  });

  it("should have valid blobs forks", () => {
    expect(forkBlobs).toMatchSnapshot();
  });

  describe("highestFork", () => {
    it("should return the only fork as highest", () => {
      expect(highestFork([ForkName.altair])).toBe(ForkName.altair);
    });

    it("should return the the highest fork", () => {
      expect(highestFork([ForkName.altair, ForkName.bellatrix])).toBe(ForkName.bellatrix);
    });

    it("should return the the highest fork if given in random order", () => {
      expect(highestFork([ForkName.altair, ForkName.bellatrix, ForkName.deneb, ForkName.phase0])).toBe(ForkName.deneb);
    });
  });

  describe("lowestFork", () => {
    it("should return the only fork as lowest", () => {
      expect(lowestFork([ForkName.altair])).toBe(ForkName.altair);
    });

    it("should return the the lowest fork", () => {
      expect(lowestFork([ForkName.altair, ForkName.bellatrix])).toBe(ForkName.altair);
    });

    it("should return the the lowest fork if given in random order", () => {
      expect(lowestFork([ForkName.altair, ForkName.bellatrix, ForkName.deneb, ForkName.phase0])).toBe(ForkName.phase0);
    });
  });
});
