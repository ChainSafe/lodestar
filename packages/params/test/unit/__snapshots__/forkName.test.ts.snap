// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`forkName > should have valid allForks 1`] = `
[
  "phase0",
  "altair",
  "bellatrix",
  "capella",
  "deneb",
  "electra",
]
`;

exports[`forkName > should have valid blobs forks 1`] = `
[
  "deneb",
  "electra",
]
`;

exports[`forkName > should have valid execution forks 1`] = `
[
  "bellatrix",
  "capella",
  "deneb",
  "electra",
]
`;

exports[`forkName > should have valid lightclient forks 1`] = `
[
  "altair",
  "bellatrix",
  "capella",
  "deneb",
  "electra",
]
`;

exports[`forkName > should have valid withdrawal forks 1`] = `
[
  "capella",
  "deneb",
  "electra",
]
`;
